src/hal/src/pso/descriptor.rs

```diff
@@ -67,6 +67,91 @@ pub enum DescriptorType {
     InputAttachment = 10,
 }
 
+impl std::convert::From<DescriptorType> for RawDescriptorType {
+    fn from(ty: DescriptorType) -> Self {
+        match ty {
+            DescriptorType::Sampler => RawDescriptorType::Sampler,
+            DescriptorType::CombinedImageSampler => RawDescriptorType::CombinedImageSampler,
+            DescriptorType::Image { ty } => match ty {
+                ImageDescriptorType::Storage => RawDescriptorType::StorageImage,
+                ImageDescriptorType::Sampled => RawDescriptorType::SampledImage,
+            }
+            DescriptorType::Buffer { access, format } => match access {
+                BufferDescriptorAccess::Storage => match format {
+                    BufferDescriptorFormat::Dynamic => RawDescriptorType::StorageBufferDynamic,
+                    BufferDescriptorFormat::Structured => RawDescriptorType::StorageBuffer,
+                    BufferDescriptorFormat::Texel => RawDescriptorType::StorageTexelBuffer,
+                },
+                BufferDescriptorAccess::Uniform => match format {
+                    BufferDescriptorFormat::Dynamic => RawDescriptorType::UniformBufferDynamic,
+                    BufferDescriptorFormat::Structured => RawDescriptorType::UniformBuffer,
+                    BufferDescriptorFormat::Texel => RawDescriptorType::UniformTexelBuffer,
+                },
+            }
+            DescriptorType::InputAttachment => RawDescriptorType::InputAttachment,
+        }
+    }
+}
+
+/// Access type of a buffer.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum BufferDescriptorAccess {
+    /// Storage buffers allow load, store, and atomic operations.
+    Storage,
+    /// Uniform buffers allow only load operations.
+    Uniform,
+}
+
+/// Format of a buffer.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum BufferDescriptorFormat {
+    // TODO: seems to have no effect in hal, ask about removing or documenting that it's a no-op
```

Comment:
This has great effect, which we've been using for a long time, especially in wgpu.
What made you confused about this?

Reply:
I made this comment looking at the "legacy" backends (gl/dx11), realized that it behaved as expected in the modern frontends (vk/mtl/dx12) and promptly forgot to remove the comment 😅